### PR TITLE
logout endpoint  auth_plug_example App issue #15

### PR DESCRIPTION
**Pull request by nelsonic**
+ [x] Implements MVP of **logout** functionality. ⏳ #15

### CODE REVIEW

**Comment by nelsonic** [comment]
Ahh ...
image

Need ***tests*** ... 🙄 
Added them to **`auth_plug`** https://github.com/dwyl/auth_plug/pull/36 ✅ 
but need them here too ... ⏳

**Comment by nelsonic** [comment]
Good thing testing in Elixir/Phoenix is so easy. 😉 
Less than a minute later, back up to 100% coverage. 💯

**Comment by nelsonic** [comment]
The code I've added to this PR should be "enough" to logout in both the "consumer" (`auth_plug_example`) and the **`auth`** App. Once `AuthPlug.logout/1` is finished, this PR should "just work". 💭

File: lib/app_web/controllers/page_controller.ex [hunk 1]
```diff
@@ -18,4 +18,10 @@ defmodule AppWeb.PageController do
   def ping(conn, params) do
     Ping.render_pixel(conn, params)
   end
+
+  def logout(conn, _params) do
+    conn
+    |> AuthPlug.logout()
+    |> render("loggedout.html")
+  end
```
[the remark this answers is not in the document]
Comment (SimonLab) [reply lib/app_web/controllers/page_controller.ex:26]: Thanks for the example, makes sense in this case to leave it to the auth_plug user to add halt if required

**Comment by nelsonic** [comment]
@SimonLab hopefully this PR is clear. 
The idea is that we can _easily_ add the `/logout` endpoint to _any_ App using `auth_plug`.
The handler will invoke the `AuthPlug.logout/1` function implemented in https://github.com/dwyl/auth_plug/pull/36
Which in turn will make an HTTP Request to `auth_url/end_session` see: https://github.com/dwyl/auth/pull/159

Once the person has `auth_plug` working, the `/logout` functionality should be _very_ simple to implement.  💭 
Let me know if you agree. 🤞 
Feedback/improvement very much welcome!

File: lib/app_web/controllers/page_controller.ex [hunk 2]
```diff
@@ -18,4 +18,10 @@ defmodule AppWeb.PageController do
   def ping(conn, params) do
     Ping.render_pixel(conn, params)
   end
+
+  def logout(conn, _params) do
+    conn
+    |> AuthPlug.logout()
```
Comment (nelsonic) [inline lib/app_web/controllers/page_controller.ex:24]: This is the interesting part. 💡 
With this single invocation of `AuthPlug.logout/1` all "logout" action is handled transparently.

Comment (SimonLab) [reply lib/app_web/controllers/page_controller.ex:24]: I was wondering if the logout function could failed before rendering the `loggedout.html` page.
I've created an issue on auth_plug: https://github.com/dwyl/auth_plug/issues/38

**Review by SimonLab — APPROVED** [review]
LGTM :+1:

File: lib/app_web/controllers/page_controller.ex (hunk 2 above)
Comment (nelsonic) [reply lib/app_web/controllers/page_controller.ex:24]: Excellent idea to check for errors. Thanks.

**Pull request merged**